src/huggingface_hub/commands/user.py

```diff
@@ -311,42 +329,76 @@ def notebook_login():
-    password_widget = widgets.Password(description="Password:")
-    finish_button = widgets.Button(description="Login")
     box_layout = widgets.Layout(
         display="flex", flex_flow="column", align_items="center", width="50%"
     )
-    main_widget = widgets.VBox(
+
+    token_widget = widgets.Password(description="Token:")
+    token_finish_button = widgets.Button(description="Login")
+    switch_button = widgets.Button(description="Use password")
+
+    login_token_widget = widgets.VBox(
         [
-            widgets.HTML(value=LOGIN_NOTEBOOK_HTML),
+            widgets.HTML(NOTEBOOK_LOGIN_TOKEN_HTML_START),
+            token_widget,
+            token_finish_button,
+            widgets.HTML(NOTEBOOK_LOGIN_TOKEN_HTML_END),
+            switch_button,
+        ],
+        layout=box_layout,
+    )
+    display(login_token_widget)
+
+    # Deprecated page for login
+    input_widget = widgets.Text(description="Username:")
+    password_widget = widgets.Password(description="Password:")
+    password_finish_button = widgets.Button(description="Login")
+
+    login_password_widget = widgets.VBox(
+        [
+            widgets.HTML(value=NOTEBOOK_LOGIN_PASSWORD_HTML),
             widgets.HBox([input_widget, password_widget]),
-            finish_button,
+            password_finish_button,
         ],
         layout=box_layout,
     )
 
-    display(main_widget)
+    # On click events
+    def login_token_event(t):
+        token = token_widget.value
+        # Erase token and clear value to make sure it's not saved in the notebook.
+        token_widget.value = ""
+        clear_output()
+        _login(token=token)
+
+    token_finish_button.on_click(login_token_event)
 
-    def login_event(t):
+    def login_password_event(t):
         username = input_widget.value
         password = password_widget.value
         # Erase password and clear value to make sure it's not saved in the notebook.
         password_widget.value = ""
         clear_output()
-        _login(HfApi(), username, password)
+        _login(HfApi(), username=username, password=password)
 
-    finish_button.on_click(login_event)
+    password_finish_button.on_click(login_password_event)
 
+    def switch_event(t):
+        clear_output()
+        display(login_password_widget)
 
-def _login(hf_api, username, password):
-    try:
-        token = hf_api.login(username, password)
-    except HTTPError as e:
-        # probably invalid credentials, display error message.
-        print(e)
-        print(ANSI.red(e.response.text))
-        exit(1)
+    switch_button.on_click(switch_event)
+
+
+def _login(hf_api, username=None, password=None, token=None):
+    if token is None:
+        try:
+            token = hf_api.login(username, password)
+        except HTTPError as e:
+            # probably invalid credentials, display error message.
+            print(e)
+            print(ANSI.red(e.response.text))
+            exit(1)
     HfFolder.save_token(token)
```

Comment:
Does it make sense to call `HfApi._is_valid_token` here to ensure the passed token is valid ?